x-pack/plugins/timelines/server/search_strategy/timeline/utils/build_ecs_objects.test.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { eventHit } from '@kbn/securitysolution-t-grid';
+import { EventHit } from '../../../../common/search_strategy';
+import { buildEcsObjects } from './build_ecs_objects';
+
+describe('buildEcsObjects', () => {
+  it('should not populate empty or null ecs fields', () => {
+    const hitWithMissingInfo: EventHit = {
+      _index: '.test-index',
+      _id: 'test-id',
+      _score: 0,
+      _source: {
+        host: {
+          architecture: 'windows98',
+          hostname: 'test-name',
+          id: 'some-id',
+          ip: [],
+          name: 'test-name',
+        },
+      },
+      fields: {
+        'host.architecture': ['windows98'],
+        'host.hostname': ['test-name'],
+        'host.id': ['some-id'],
+        'host.ip': [],
+        'host.name': ['test-name'],
+      },
+      _type: '',
+      sort: ['1610199700517'],
+    };
+
+    const ecsObject = buildEcsObjects(hitWithMissingInfo);
+    expect(ecsObject).toEqual({
+      _id: 'test-id',
+      _index: '.test-index',
+      host: {
+        id: ['some-id'],
+        ip: [],
```

Comment:
is the empty `ip` array expected here given the `should not populate empty or null ecs fields` title of the test?

Reply:
Perhaps just the test name `should not populate empty or null ecs fields` is misleading, because per the screenshot below, a field containing an empty array in `_source` still shows up in the table view, and as a user, it makes sense to see that empty-but-populated field:

<img width="1149" alt="field_with_empty_array" src="https://user-images.githubusercontent.com/4459398/145336332-ba6e7fcc-f3fe-456f-af4d-b32dcc3ce514.png">

Reply:
The description was wrong, I've updated. Thank you!